src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -528,6 +533,25 @@ export class ChatWidget extends Disposable implements IChatWidget {
 	setDynamicChatTreeItemLayout(numOfChatTreeItems: number, maxHeight: number) {
 		this._dynamicMessageLayoutData = { numOfMessages: numOfChatTreeItems, maxHeight };
 		this._register(this.renderer.onDidChangeItemHeight(() => this.layoutDynamicChatTreeItemMode()));
+
+		const mutableDisposable = this._register(new MutableDisposable());
+		this._register(this.tree.onDidScroll((e) => {
+			mutableDisposable.value = dom.scheduleAtNextAnimationFrame(() => {
+				mutableDisposable.dispose();
+				if (!e.scrollTopChanged || e.heightChanged || e.scrollHeightChanged) {
+					return;
+				}
+				const renderHeight = e.height;
+				const diff = e.scrollHeight - renderHeight - e.scrollTop;
+				if (diff === 0) {
+					return;
+				}
+
+				const newHeight = Math.min(renderHeight + diff, maxHeight);
+				const inputPartHeight = this.inputPart.layout(newHeight, this.container.offsetWidth);
+				this.layout(newHeight + inputPartHeight, this.container.offsetWidth);
```

Comment:
Might be nice to cache the width (is it just some constant?) just to avoid reading the DOM here.

Reply:
Added this in:
https://github.com/microsoft/vscode/pull/191432

we already cache it :)